src/basic/parse-util.c

```diff
@@ -747,3 +747,34 @@ int parse_loadavg_fixed_point(const char *s, loadavg_t *ret) {
 
         return store_loadavg_fixed_point(i, f, ret);
 }
+
+static bool nft_first_char_valid(const char c) {
+        return (c >= 'a' && c <= 'z') ||
+                (c >= 'A' && c <= 'Z');
+}
+
+static bool nft_next_char_valid(const char c) {
+        return (c >= 'a' && c <= 'z') ||
+                (c >= 'A' && c <= 'Z') ||
+                (c >= '0' && c <= '9') ||
+                c == '/' || c == '\\' || c == '_' || c == '.';
+}
+
+/* Limitations are described in https://www.netfilter.org/projects/nftables/manpage.html and
+ * https://bugzilla.netfilter.org/show_bug.cgi?id=1175 */
+bool nft_identifier_valid(const char *id) {
+        assert(id);
```

Comment:
doesn't matter much, but in newer code that validates strings we also followed the rule that passing NULL to it returns "false", i.e. a NULL pointer is never considered a valid string